libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -553,22 +450,16 @@ void UARTDriver::_timer_tick(void)
     }
 
     // try to fill the read buffer
-    uint16_t _head;
-    n = BUF_SPACE(_readbuf);
-    if (n > 0) {
-        uint16_t n1 = _readbuf_size - _readbuf_tail;
-        if (n1 >= n) {
-            // one read will do
-            assert(_readbuf_tail+n <= _readbuf_size);
-            _read_fd(&_readbuf[_readbuf_tail], n);
-        } else {
-            assert(_readbuf_tail+n1 <= _readbuf_size);
-            int ret = _read_fd(&_readbuf[_readbuf_tail], n1);
-            if (ret == n1 && n > n1) {
-                assert(_readbuf_tail+(n-n1) <= _readbuf_size);
-                _read_fd(&_readbuf[_readbuf_tail], n - n1);
-            }
-        }
+    int ret;
+    ByteBuffer::IoVec vec[2];
+
+    const auto n_vec = _readbuf.reserve(vec, _readbuf.space());
+    for (int i = 0; i < n_vec; i++) {
+        ret = _read_fd(vec[i].data, vec[i].len);
+        assert(ret < 0);
```

Comment:
Although the assert seems wrong, the removal of it doesn't really help. The console is getting polluted with messages like
```Info: GPS 0: detected as NMEA at 9600 baud```
```Info: GPS 0: detected as u-blox at 115200 baud```
(the list goes on and on)

So that was kind of a good thing we had this assert in there.